openff/models/types.py

```diff
@@ -56,17 +56,22 @@ def validate_type(cls, val):
                     # some custom behavior could go here
                     assert unit_.dimensionality == val.dimensionality
                     # return through converting to some intended default units (taken from the class)
+                    val = float(val.m) * val.u
```

Comment:
```suggestion
                    val._magnitude = float(val.m)
```

It'd be quite a bit faster to forcefully convert this:

```python
In [6]: %%timeit
   ...: val._magnitude = float(val.m)
   ...:
   ...:
157 ns ± 0.483 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [7]: %%timeit
   ...: float(val.m) * val.u
   ...:
   ...:
11.6 µs ± 35.2 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

obviously this could break everything horribly as well. Not sure